src/Static.jl

```diff
@@ -85,14 +87,14 @@ Returns `True` if `T` is a static type.
 
 See also: [`static`](@ref), [`known`](@ref)
 """
-is_static
-@aggressive_constprop is_static(x) = is_static(typeof(x))
-is_static(::Type{T}) where {T<:StaticInt} = True()
-is_static(::Type{T}) where {T<:StaticBool} = True()
-is_static(::Type{T}) where {T<:StaticSymbol} = True()
-is_static(::Type{T}) where {T<:Val} = True()
-is_static(::Type{T}) where {T} = False()
-is_static(::Type{T}) where {T<:StaticFloat64} = True()
+is_static(@nospecialize(x)) = is_static(typeof(x))
+is_static(@nospecialize(x::Type{<:StaticInt})) = True()
+is_static(@nospecialize(x::Type{<:StaticBool})) = True()
+is_static(@nospecialize(x::Type{<:StaticSymbol})) = True()
+is_static(@nospecialize(x::Type{<:Val})) = True()
+is_static(@nospecialize(x::Type{<:StaticFloat64})) = True()
+is_static(x::Type{T}) where {T} = False()
+
```

Comment:
@timholy , If I want to avoid generating a unique method for every variant of `StaticInt` is this the right approach or will it kill inference somewhere down the line?

Reply:
Presumably everything still infers and optimizes fine, because the behavior doesn't actually depend on the particular types?

Reply:
That's the goal. I was hoping to start strategically adding these to trait functions so that we can avoid some of the pitfalls of StaticArrays.